### PR TITLE
Rework onFragmentFirstTimeVisible, use isVisible&isResumed

### PR DESCRIPTION
**Pull request by harshad1**
Using isVisible() and isResumed() now. Better calls with `post`

### CODE REVIEW

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java [hunk 1]
```diff
@@ -149,17 +144,20 @@ public String getAppLanguage() {
...
         }
+    }
 
-        attachToolbarClickListenersToFragment();
+    @Override
+    public void onResume() {
+        super.onResume();
+        final View view = getView();
+        if (view != null) {
+            view.postDelayed(this::checkRunFirstTimeVisible, 1);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java:159]: May be good to bump up the delay to something higher. Something like ~ 200ms will be barely perceptible to user, but everything will have had time to draw and layout.

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java [hunk 2]
```diff
@@ -70,12 +69,7 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
...
         view.postDelayed(() -> {
-            synchronized (_fragmentFirstTimeVisibleSync) {
-                if (getUserVisibleHint() && isVisible() && _fragmentFirstTimeVisible) {
-                    _fragmentFirstTimeVisible = false;
-                    onFragmentFirstTimeVisible();
-                }
-            }
+            checkRunFirstTimeVisible();
             attachToolbarClickListenersToFragment();
         }, 1);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java:74]: Should probably only call from onResume - that is when the fragment is fully ready and displayed

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java [hunk 3]
```diff
@@ -149,17 +128,33 @@ public String getAppLanguage() {
...
+            onFragmentFirstTimeVisible();
+            attachToolbarClickListenersToFragment();
         }
+    }
 
-        attachToolbarClickListenersToFragment();
+    protected void attachToolbarClickListenersToFragment() {
+        final Toolbar toolbar = getToolbar();
+        if (toolbar != null) {
+            toolbar.setOnLongClickListener(clickView -> isVisible() && isResumed() && onToolbarLongClicked(clickView));
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java:142]: suggestion: Callback.b2<bool,view> (islong, view) {

if ( isvisible && isRes.....) {


  if islong { return ontoolbarlongclick() }

  else { ontoolbarclick }
}
return false

---


setOnClivkListener(v -> callback(false, v));

setOnLonglivkListener(v -> callback(true, v));

what do you think?

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java (hunk 1 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java]: stability over speed nitpickers :-)

Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java]: It's still blazing fast. I have not found an editor which opens and highlights a large file as fast as markor :)

Comment (gsantner) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java]: Can you remotly reproduce the permoance issue mentioned in #1833 with ~5 seconds? I do not.

Especially with the permission request change it got much quicker on first start and further starts as far I can tell. (unrelated to edior)

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java (hunk 3 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java:142]: IMO it's extra indirection for no real benefit.

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java (hunk 1 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java]: > Can you remotly reproduce the permoance issue mentioned in https://github.com/gsantner/markor/issues/1833 with ~5 seconds? I do not.

My reading was that this is happening with the debug build... Which I expect to be much slower/

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java (hunk 3 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java:142]: its easier to debug as you can put a break point on i.e. isVisible line , to add i.e. another condition to both clicks

File: app/src/main/java/net/gsantner/markor/format/ActionButtonBase.java [hunk 4]
```diff
@@ -596,7 +596,7 @@ protected final boolean runCommonAction(final @StringRes int action) {
                 return true;
             }
             case R.string.abid_common_ordered_list_renumber: {
-                renumberOrderedList(TextViewUtils.getSelection(_hlEditor)[0]);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/ActionButtonBase.java:599]: So I realized that selection is a property of the editable, not the textview. This means we can simplify the call to renumber and preserve selection

File: app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java [hunk 5]
```diff
@@ -116,24 +119,33 @@ private void updateDynamicHighlighting() {
...
 
+    private synchronized void postUpdateDynamicHighlighting() {
+        if (!_hlPostQueued) {
+            _hlPostQueued = true;
+            post(() -> {
+                updateDynamicHighlighting();
+                _hlPostQueued = false;
+            });
+        }
+    }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java:130]: Using the post prevents the selection from being hidden when the keyboard is shown

**Pull request merged**